dashboard/src/api/hooks/useDeleteVersion.ts

```diff
@@ -0,0 +1,29 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { GetVersionsResponse } from '../../interfaces/versions';
+import client from '../client';
+import '../mocks/useDeleteVersion.mock';
+
+const deleteVersion = async (projectId?: number, versionId?: number) => {
```

Comment:
The props are optional. there is no need to call the server is we have undefined. You might be checking if projectid or versionid is not undefined before calling this function, but then you should remove the ?.

Reply:
yeah totally makes sense